pkg/kubefedctl/enable/validation.go

```diff
@@ -171,6 +171,61 @@ func ValidationSchema(specProps v1beta1.JSONSchemaProps) *v1beta1.CustomResource
 					Type: "object",
 				},
 				"spec": specProps,
+				"status": {
+					Type: "object",
+					Properties: map[string]v1beta1.JSONSchemaProps{
+						"conditions": {
```

Comment:
This is what I was trying to understand in the sync today.

So this looks like that the fields are directly under status which can lead to restructuring the API if we add something similar in hierarchy to propagation status in future.
Would it make any sense to say have it like below:
```
status:
  propagationStatus:
    conditions:
    .......
``` 

To allow a possibility of something like below in future:
```
status:
  propagationStatus:
    conditions:
    .......
  aggregatedStatus:
    .......
``` 
I am not implying that we need to do any change now. I am also not aware of any use case needing all statuses for all objects from all clusters.
Additionally please correct me if I am wrong or assuming something.

Reply:
I don't think aggregated status will ever belong on a federated resource.  The kube api is a configuration store for the agents that operate on a cluster, not a general-purpose database for storing arbitrary amounts of data.  It can certainly be (and is) done, but I don't think we should directly support doing so on primary resources like federated types.

Are there reasons other than the potential for including aggregated status to suggest adding the status fields proposed by this PR under the `propagationStatus` key?

Reply:
None as of now.